app/lib/features/pins/actions/reduct_pin_action.dart

```diff
@@ -0,0 +1,22 @@
+// redact pin dialog
+import 'package:acter/common/actions/redact_content.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+
+void showRedactDialog({
+  required BuildContext context,
+  required ActerPin pin,
```

Comment:
This feel like useless to me when the only required object is ```eventId``` string. And its making ```redactDialog``` redundant...

Reply:
Haven't did any changes in this code apart from moving it from pin-detail page to action file.

This is the same code used in the event, task, pin, update etc. Would love to do this whole changes as separate PR to minimised complexity of this PR.